packages/@aws-cdk/aws-glue/lib/database.ts

```diff
@@ -83,27 +82,29 @@ export class Database extends Resource implements IDatabase {
   /**
    * Location URI of this database.
    */
-  public readonly locationUri: string;
+  public readonly locationUri?: string;
 
   constructor(scope: Construct, id: string, props: DatabaseProps) {
     super(scope, id, {
       physicalName: props.databaseName,
     });
 
+    let databaseInput: CfnDatabase.DatabaseInputProperty = {
+      name: props.databaseName,
+    };
+
     if (props.locationUri) {
```

Comment:
This should probably be `props.locationUri !== undefined` instead, to correctly handle the empty location case (not sure if it's valid, but probably should be possible).

Reply:
The idea is actually to exclude `locationUri` if it is an empty string.
Aws does not allow empty strings there. See https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-catalog-databases.html 
```
LocationUri – Uniform resource identifier (uri), not less than 1 or more than 1024 bytes long...
```
So we have to exclude uri from the `databaseInput`. This is what `if (props.locationUri)` (empty string is false in js). But I can make it less magical by writing smth like
```
if(props.locationUri !== undefined && props.locationUri.length >= 1) {...}
```

Reply:
Then we should fail with an exception if the passed `locationUri` is empty. Would you mind doing that (+ a unit test for this behavior)?

Reply:
@skinny85 Showing an exception makes sense. Updated the pr, and removed the mention about the s3 from the readme file.